catalog/imaging/imaging-plugin-nitf/src/test/java/org/codice/alliance/plugin/nitf/PostIngestPluginTest.java

```diff
@@ -0,0 +1,214 @@
+    nitfPostIngestPlugin.process((CreateResponse) null);
+  }
+
+  @Test(expected = PluginExecutionException.class)
+  public void testNullInputOnUpdate() throws PluginExecutionException {
+    nitfPostIngestPlugin.process((UpdateResponse) null);
+  }
+
+  @Test
+  public void testCreateResponse() throws Exception {
+    nitfPostIngestPlugin.process(createResponse);
+    validate();
+  }
+
+  @Test
+  public void testTooLargeCreateResponse() throws PluginExecutionException {
+    nitfPostIngestPlugin.setMaxNitfSizeMB(0);
+    nitfPostIngestPlugin.process(createResponse);
+    assertThat(metacard.getThumbnail(), is(nullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(nullValue()));
+  }
+
+  @Test
+  public void testCreateResponseNoOverview() throws PluginExecutionException {
+    nitfPostIngestPlugin.setCreateOverview(false);
+    nitfPostIngestPlugin.process(createResponse);
+    assertThat(metacard.getThumbnail(), is(notNullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(notNullValue()));
+    assertThat(
+        metacard.getAttribute(Core.DERIVED_RESOURCE_URI).getValue().toString(),
+        containsString("original"));
+  }
+
+  @Test
+  public void testCreateResponseNoOriginal() throws PluginExecutionException {
+    nitfPostIngestPlugin.setStoreOriginalImage(false);
+    nitfPostIngestPlugin.process(createResponse);
+    assertThat(metacard.getThumbnail(), is(notNullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(notNullValue()));
+    assertThat(
+        metacard.getAttribute(Core.DERIVED_RESOURCE_URI).getValue().toString(),
+        containsString("overview"));
+  }
+
+  @Test
+  public void testUpdateResponse() throws Exception {
+    nitfPostIngestPlugin.process(updateResponse);
+    validate();
+  }
+
+  @Test
+  public void testNonNitfContent() throws Exception {
+    metacard.setAttribute(new AttributeImpl(Media.TYPE, "text/xml"));
+    nitfPostIngestPlugin.process(updateResponse);
+    assertThat(metacard.getThumbnail(), is(nullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(nullValue()));
+  }
+
+  @Test
+  public void testNoMediaType() throws Exception {
+    metacard.setAttribute(Media.TYPE, null);
+    nitfPostIngestPlugin.process(updateResponse);
+    assertThat(metacard.getThumbnail(), is(nullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(nullValue()));
+  }
+
+  @Test
+  public void testAlreadyProcessedRequest() throws Exception {
+    requestProperties.put("NitfPostIngestPlugin.Processing", true);
+    nitfPostIngestPlugin.process(createResponse);
+    assertThat(metacard.getThumbnail(), is(nullValue()));
+    assertThat(metacard.getAttribute(Core.DERIVED_RESOURCE_URI), is(nullValue()));
+  }
+
+  private void validate() throws Exception {
+    verify(catalogFramework, times(1)).update(any(UpdateStorageRequest.class));
```

Comment:
I think these uit tests should also test that the right request was made to the catalog framework. That is, that the `ContentItem`s in the`UpdateStorageRequest` have the right qualifiers.